internal/repo/datasource/sql_test.go

```diff
@@ -0,0 +1,357 @@
+package datasource
```

Comment:
This layer is fully mocked and thus should have 100% code coverage.

Reply:
there is one part that is not getting mocked when first rows.scan is called

Reply:
done

Reply:
From the test cases, looks like this may not be 100% covered. Re-check and paste screenshot of 100% coverage for the `datasource` layer.